voc/python/ast.py

```diff
@@ -325,8 +325,79 @@ def visit_Delete(self, node):
 
     @node_visitor
     def visit_Assign(self, node):
-        # Evaluate the value
-        self.visit(node.value)
+        def push_message(node):
```

Comment:
This shouldn't be an closure method - it's not dependent on any of the context provided by `visit_Assign`. You can move it into a utility method at the level of the visitor (like e.g., `full_classspec`).

Reply:
Yes. Completely agree with that, I realized expression is more than just assignment and augmented assignment. I'm currently working on turning the block of code into an utility function which can be called in any node that evaluate an expression.